taichi/transforms/simplify.cpp

```diff
@@ -705,10 +705,9 @@ class BasicBlockSimplify : public IRVisitor {
       throw IRModified();
     }
 
-    // step 1: eliminate useless extraction of another OffsetAndExtractBitsStmt
-    if (stmt->offset == 0 && stmt->bit_begin == 0 &&
-        stmt->input->is<OffsetAndExtractBitsStmt>()) {
-      auto bstmt = stmt->input->as<OffsetAndExtractBitsStmt>();
+    // step 1: eliminate useless extraction of another BitExtractStmt
+    if (stmt->bit_begin == 0 && stmt->input->is<BitExtractStmt>()) {
+      auto bstmt = stmt->input->as<BitExtractStmt>();
       if (stmt->bit_end == bstmt->bit_end - bstmt->bit_begin) {
```

Comment:
```suggestion
      if (stmt->bit_end >= bstmt->bit_end - bstmt->bit_begin) {
```
(Off-topic but maybe we can quickly fix here. Not sure if it makes sense.)